napi-inl.h

```
@@ -165,7 +165,7 @@ napi_value TemplatedInstanceCallback(napi_env env,
   return details::WrapCallback([&] {
     CallbackInfo cbInfo(env, info);
     T* instance = T::Unwrap(cbInfo.This().As<Object>());
-    return (instance->*UnwrapCallback)(cbInfo);
+    return instance ? (instance->*UnwrapCallback)(cbInfo) : nullptr;
```

```suggestion
    return instance ? (instance->*UnwrapCallback)(cbInfo) : Napi::Value();
```

---

I would have liked this to throw instead of returning `undefined`.

I'd assume this can only happen if the `InstanceWrap` is setup or called incorrectly (i.e. doesn't have a its instance data set) and I therefore expected it to throw the same way other native APIs throw when I call an unbound member function:

```
> const b = Buffer.alloc(1);
undefined
> const copy = b.copy;
undefined
> copy();
Uncaught:
TypeError [ERR_INVALID_ARG_TYPE]: The "source" argument must be an instance of Buffer or Uint8Array. Received undefined
```